Doc/library/functools.rst

```diff
@@ -110,8 +110,31 @@ The :mod:`functools` module defines the following functions:
            def stdev(self):
                return statistics.stdev(self._data)
 
+    By default, a ``cached_property`` instance contains an
+    :class:`~threading.RLock` which prevents multiple threads from executing
+    the cached property concurrently. This was intended to ensure that the
+    cached property would execute only once per instance, but it also prevents
+    concurrent execution on different instances of the same class, causing
+    unnecessary and excessive serialization of updates. This locking behavior
+    is deprecated and can be avoided by passing ``lock=False``::
+
+        class DataSet:
+            def __init__(self, sequence_of_numbers):
+                self._data = sequence_of_numbers
+
+            @cached_property(lock=False)
+            def stdev(self):
+                return statistics.stdev(self._data)
+
+    With ``lock=False``, ``cached_property`` provides no guarantee that it will
+    run only once per instance, if accessed in multiple threads. An application
+    wanting such guarantees must handle its own locking explicitly.
+
    .. versionadded:: 3.8
 
+   .. versionchanged:: 3.12
+      The ``lock`` argument was added.
+
```

Comment:
```suggestion

   .. deprecated-removed:: 3.12 3.14
      The locking behaviour of ``cached_property`` is deprecated, and will be
      removed in 3.14. Use ``lock=True`` to avoid the deprecated behaviour.
```